flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/JsonRowDataDeserializationSchema.java

```diff
@@ -130,6 +133,39 @@ public RowData deserialize(byte[] message) throws IOException {
 		}
 	}
 
+	@Override
+	public void deserialize(byte[] message, Collector<RowData> out) throws IOException {
+		try {
+			final JsonNode root = objectMapper.readTree(message);
+
+			if (root.isArray()) {
+				List<RowData> rowDataList = new ArrayList<>();
+				ArrayNode arrayNode = (ArrayNode) root;
+				for (int i = 0; i < arrayNode.size(); ++i) {
+					RowData result = (RowData) runtimeConverter.convert(arrayNode.get(i));
+					if (result != null) {
+						rowDataList.add(result);
+					}
+				}
+
+				// collect final results in a separate for-loop in case of parse errors.
```

Comment:
I think the parse errors can still be catched if we collect each parsed row eagerly.